protocols/bgp/packet/path_attributes_test.go

```diff
@@ -1479,6 +1632,106 @@ func TestSerializeCommunities(t *testing.T) {
 	}
 }
 
+func TestSerializeOriginatorID(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       *PathAttribute
+		expected    []byte
+		expectedLen uint8
+	}{
+		{
+			name: "Valid OriginatorID",
+			input: &PathAttribute{
+				TypeCode: OriginatorIDAttr,
+				Value:    bnet.IPv4FromOctets(1, 1, 1, 1).ToUint32(),
+			},
+			expected: []byte{
+				setOptional(uint8(0)), // Attribute flags
+				OriginatorIDAttr,      // Type
+				4,                     // Length
+				1, 1, 1, 1,            // ClusterID (1.1.1.1)
+			},
+			expectedLen: 7,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(te *testing.T) {
+			buf := bytes.NewBuffer(nil)
+			n := test.input.serializeOriginatorID(buf)
+			if n != test.expectedLen {
+				t.Errorf("Unexpected length for test %q: %d", test.name, n)
+			}
+
+			assert.Equal(t, test.expected, buf.Bytes())
```

Comment:
pass te instead of t here, otherwise all tests fail when one test case fails. when we start parallelising tests this would hurt us. better rename te to t, then the outer t is hidden